classlib/src/main/java/org/teavm/classlib/java/util/TBitSet.java

```diff
@@ -76,12 +86,19 @@ public static TBitSet valueOf(byte[] bytes) {
         return new TBitSet(ints);
     }
 
+    public static TBitSet valueOf(TByteBuffer buff) {
+        buff = buff.slice().order(TByteOrder.LITTLE_ENDIAN);
+        byte[] words = new byte[buff.remaining()];
+        buff.get(words);
+        return valueOf(words);
+    }
+
     public byte[] toByteArray() {
         byte[] bytes = new byte[(length + 7) / 8];
-        int fullInts = length / TInteger.SIZE;
+        int fullInts = bytes.length / 4;
```

Comment:
Why did you replace `TInteger.SIZE` with `4`?

Reply:
In case when you have length = 25, you have an issue. THis change removes the issue. It should be either (length + 7) / 32 or bytes.length / 4. I used second because we already calculated bytes.length.